zerver/views/messages.py

```diff
@@ -264,15 +264,20 @@ def by_pm_with(self, query, operand, maybe_negate):
 
     def by_search(self, query, operand, maybe_negate):
         # type: (Query, str, ConditionTransform) -> Query
+        rendered_operand = render_markdown(None, operand)
+        for remove in ('<p>', '</p>', '<br>'):
+            rendered_operand = rendered_operand.replace(remove, '')
```

Comment:
I see, this is to prevent highlighting the entire string when the only thing there is a <p> tag.  Makes sense.